### ISSUE TITLE
In Docs, "Configuration Based on Glob Patterns" should specify it's a v4.1 feature. Fails silently on earlier versions.

### ISSUE THREAD

**Issue by geoffreyyip** [eslint/eslint#8793]
**Tell us about your environment**

* **ESLint Version: 3.x**
* **Node Version: 7.x**
* **npm Version: 5.x**

**Problem:**

I had a default configuration for my `**/*.js` files and I was trying to set linting rules differently for my `**/*.test.js` files. Mostly, I wanted to use `describe`, `it`, and other global testing variables in my testing files. I was trying to avoid messages like `error    'describe' is not defined       no-undef`

On the Configuring ESLint's ["Configuration Based on Glob Patterns"](http://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns) section, I found out I could specify glob-based overrides, and so I specified my configuration to something like this.

`eslintrc.js`
```javascript
{
  "env": { "node": true },
  "overrides": [
    {
      "files": [ "**/*.test.js" ],
      "env": { "mocha": true }
    }
  ]
}
``` 

**What did you expect to happen?**

I expected my `*.test.js` files to be linted differently. I expected to get rid of the `error    'describe' is not defined       no-undef` message. Instead, it persisted. And I had no idea why. I spent a hour checking my globs and configuration before realizing that I was still on v3.19.0 and the glob-based overrides were only added 2 days ago, according to #3611.

Once I upgraded to v4.1.0, the `overrides` syntax worked as normal.

Two things that could help future developers avoid my mistake:
* A note in the docs saying users should upgrade to v4.1.0 to use the glob-based configuration feature and that it will have no effect in earlier versions.
* A CLI warning for v3.x telling users that `overrides` syntax is unavailable for them.

**Comment by ilyavolodin** [comment]
That's a good point, although we've never done that for other global features. Not sure how to best handle this.

**Comment by geoffreyyip** [comment]
[Vue.js docs](https://vuejs.org/v2/guide/components.html#sync-Modifier) handles this pretty nicely.

![version](https://user-images.githubusercontent.com/15312106/27511825-d33b8434-58fb-11e7-82ed-4be56d27aafa.PNG)

It could be a good practice going forward. `eslint` has a crazy amount of configurations, feature, and partner libraries. It's not always easy to pinpoint the bug.

Thanks for hearing me out!

**Comment by jshado1** [comment]
I'm experiencing this on v4.18
```json
"overrides": [
  {
    "env": {
      "mocha": true,
    },
    "files": [
      "**/*.spec.js"
    ],
    "rules": {
      "func-names": "off",
      "no-unused-expressions": "off",
      "prefer-arrow-callback": [
        "error",
        "never"
      ]
    }
  }
],
```

Nothing from this override block is getting applied to my spec files.

**Comment by platinumazure** [comment]
@jshado1 Please open a new issue, thanks!

**Comment by jshado1** [comment]
@platinumazure #10136

**Comment by nzakas** [comment]
It seems simple enough to add this note in the docs. I'll add it.
